subprojects/model-groovy/src/integTest/groovy/org/gradle/cli/TaskOptionsSpec.groovy

```diff
@@ -0,0 +1,104 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.cli
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+
+class TaskOptionsSpec extends AbstractIntegrationSpec {
+    def "can use a built-in option"() {
+        when:
+        buildScript """
+            abstract class MyTask extends DefaultTask {
+                @Option(option="profile", description="foobar")
+                void setProfile(String value) {
+                    System.out.println("profile=" + value)
+                }
+            }
+
+            tasks.register('mytask', MyTask.class)
+        """
+
+        then:
+        succeeds "mytask", "--profile"
+    }
+
+    def "can use a task option"() {
+        when:
+        buildScript """
+            abstract class MyTask extends DefaultTask {
+                @Option(option="profile", description="foobar")
+                @Optional
+                @Input
+                abstract Property<String> getProfile()
+
+                @TaskAction
+                void run() {
+                    if (getProfile().isPresent()) {
+                        logger.lifecycle("profile=" + getProfile().get())
+                    }
+                }
+            }
+
+            tasks.register('mytask', MyTask.class)
+        """
+
+        then:
+        succeeds "--", "mytask", "--profile", "myvalue"
+        output.contains "profile=myvalue"
+    }
+
+    def "task options apply to most recent task"() {
+        when:
+        buildScript """
+            abstract class MyTaskA extends DefaultTask {
+                @Option(option="profile", description="foobar")
+                @Optional
+                @Input
+                abstract Property<String> getProfile()
+
+                @TaskAction
+                void run() {
+                    if (getProfile().isPresent()) {
+                        logger.lifecycle("profile=" + getProfile().get())
+                    }
+                }
+            }
+
+            abstract class MyTaskB extends DefaultTask {
+                @Option(option="profile", description="foobar")
+                @Optional
+                @Input
+                abstract Property<String> getProfile()
+
+                @TaskAction
+                void run() {
+                    if (getProfile().isPresent()) {
+                        logger.lifecycle("profile=" + getProfile().get())
+                    }
+                }
+            }
+
+            tasks.register('mytaskA', MyTaskA.class)
+            tasks.register('mytaskB', MyTaskB.class)
+        """
+
+        then:
+        succeeds "--", "mytaskA", "--profile", "myvalueA", "mytaskB", "--profile", "myvalueB"
```

Comment:
I hadn't thought of this case, it's a good test to have.  I think we should also have ones to test the command lines:

`-- mytaskA mytaskB --profile myvalue`
`-- mytaskA --profile myvalue mytaskB`

results in only task B and A respectively receiving the profile value.  

We also should add one to check that this command line:

`--profile -- mytask --profile myvalue`

Runs the built-in option **and** the task option.

Reply:
Good call. These ideas likely justify "getName()" in the above comments.